### PR TITLE
[SPARK 137] Docker config updates

### PR DESCRIPTION
**Pull request by theunkn0wn1**
Low priority, but this PR updates the docker configurations to add IRC, anope, and DB to the docker-compose.

This PR also updates the dockerfile builds to 3.7.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -89,4 +89,24 @@ services:
...
+    command: ["pipenv","run","python","main.py","--config", "live.toml"]
+
+  irc:
+    image: theunkn0wn1/unrealircd
+    ports:
+      - "6667:6667"
+      - "6697:6697"
+
+  anope:
+    image: anope_services
```
Comment (Eearslya) [inline docker-compose.yml:101]: [Error] With no user qualifier, this fails on any machine but your own.

**Review by Eearslya — CHANGES_REQUESTED** [review]
These docker images appear to be flawed, as they cannot be built. (They couldn't be built before, either, but still.)

```
Step 7/8 : RUN pipenv install
 ---> Running in 06fe6bb52537
Creating a virtualenv for this project…
Pipfile: /mechasqueak/Pipfile
Using /usr/local/bin/python (3.7.3) to create virtualenv…
⠼ Creating virtual environment...Already using interpreter /usr/local/bin/python
Using base prefix '/usr/local'
New python executable in /root/.local/share/virtualenvs/mechasqueak-BtyJ3x8r/bin/python
Installing setuptools, pip, wheel...
done.
                                                                                                                                                             ✔ Successfully created virtual environment!
Virtualenv location: /root/.local/share/virtualenvs/mechasqueak-BtyJ3x8r
Installing dependencies from Pipfile.lock (993822)…
Ignoring colorama: markers 'sys_platform == "win32"' don't match your environment
Ignoring idna-ssl: markers 'python_version < "3.7"' don't match your environment
An error occurred while installing psycopg2-binary==2.7.7 --hash=sha256:19a2d1f3567b30f6c2bb3baea23f74f69d51f0c06c2e2082d0d9c28b0733a4c2 --hash=sha256:2b69cf4b0fa2716fd977aa4e1fd39af6110eb47b2bb30b4e5a469d8fbecfc102 --hash=sha256:2e952fa17ba48cbc2dc063ddeec37d7dc4ea0ef7db0ac1eda8906365a8543f31 --hash=sha256:348b49dd737ff74cfb5e663e18cb069b44c64f77ec0523b5794efafbfa7df0b8 --hash=sha256:3d72a5fdc5f00ca85160915eb9a973cf9a0ab8148f6eda40708bf672c55ac1d1 --hash=sha256:4957452f7868f43f32c090dadb4188e9c74a4687323c87a882e943c2bd4780c3 --hash=sha256:5138cec2ee1e53a671e11cc519505eb08aaaaf390c508f25b09605763d48de4b --hash=sha256:587098ca4fc46c95736459d171102336af12f0d415b3b865972a79c03f06259f --hash=sha256:5b79368bcdb1da4a05f931b62760bea0955ee2c81531d8e84625df2defd3f709 --hash=sha256:5cf43807392247d9bc99737160da32d3fa619e0bfd85ba24d1c78db205f472a4 --hash=sha256:676d1a80b1eebc0cacae8dd09b2fde24213173bf65650d22b038c5ed4039f392 --hash=sha256:6b0211ecda389101a7d1d3df2eba0cf7ffbdd2480ca6f1d2257c7bd739e84110 --hash=sha256:79cde4660de6f0bb523c229763bd8ad9a93ac6760b72c369cf1213955c430934 --hash=sha256:7aba9786ac32c2a6d5fb446002ed936b47d5e1f10c466ef7e48f66eb9f9ebe3b --hash=sha256:7c8159352244e11bdd422226aa17651110b600d175220c451a9acf795e7414e0 --hash=sha256:945f2eedf4fc6b2432697eb90bb98cc467de5147869e57405bfc31fa0b824741 --hash=sha256:96b4e902cde37a7fc6ab306b3ac089a3949e6ce3d824eeca5b19dc0bedb9f6e2 --hash=sha256:9a7bccb1212e63f309eb9fab47b6eaef796f59850f169a25695b248ca1bf681b --hash=sha256:a3bfcac727538ec11af304b5eccadbac952d4cca1a551a29b8fe554e3ad535dc --hash=sha256:b19e9f1b85c5d6136f5a0549abdc55dcbd63aba18b4f10d0d063eb65ef2c68b4 --hash=sha256:b664011bb14ca1f2287c17185e222f2098f7b4c857961dbcf9badb28786dbbf4 --hash=sha256:bde7959ef012b628868d69c474ec4920252656d0800835ed999ba5e4f57e3e2e --hash=sha256:cb095a0657d792c8de9f7c9a0452385a309dfb1bbbb3357d6b1e216353ade6ca --hash=sha256:d16d42a1b9772152c1fe606f679b2316551f7e1a1ce273e7f808e82a136cdb3d --hash=sha256:d444b1545430ffc1e7a24ce5a9be122ccd3b135a7b7e695c5862c5aff0b11159 --hash=sha256:d93ccc7bf409ec0a23f2ac70977507e0b8a8d8c54e5ee46109af2f0ec9e411f3 --hash=sha256:df6444f952ca849016902662e1a47abf4fa0678d75f92fd9dd27f20525f809cd --hash=sha256:e63850d8c52ba2b502662bf3c02603175c2397a9acc756090e444ce49508d41e --hash=sha256:ec43358c105794bc2b6fd34c68d27f92bea7102393c01889e93f4b6a70975728 --hash=sha256:f4c6926d9c03dadce7a3b378b40d2fea912c1344ef9b29869f984fb3d2a2420b! Will try again.
Ignoring pyreadline: markers 'sys_platform == "win32"' don't match your environment
Ignoring typing-extensions: markers 'python_version < "3.7"' don't match your environment
Installing initially failed dependencies…
[pipenv.exceptions.InstallError]:   File "/usr/local/lib/python3.7/site-packages/pipenv/core.py", line 1874, in do_install
[pipenv.exceptions.InstallError]:       keep_outdated=keep_outdated
[pipenv.exceptions.InstallError]:   File "/usr/local/lib/python3.7/site-packages/pipenv/core.py", line 1253, in do_init
[pipenv.exceptions.InstallError]:       pypi_mirror=pypi_mirror,
[pipenv.exceptions.InstallError]:   File "/usr/local/lib/python3.7/site-packages/pipenv/core.py", line 859, in do_install_dependencies
[pipenv.exceptions.InstallError]:       retry_list, procs, failed_deps_queue, requirements_dir, **install_kwargs
[pipenv.exceptions.InstallError]:   File "/usr/local/lib/python3.7/site-packages/pipenv/core.py", line 763, in batch_install
[pipenv.exceptions.InstallError]:       _cleanup_procs(procs, not blocking, failed_deps_queue, retry=retry)
[pipenv.exceptions.InstallError]:   File "/usr/local/lib/python3.7/site-packages/pipenv/core.py", line 681, in _cleanup_procs
[pipenv.exceptions.InstallError]:       raise exceptions.InstallError(c.dep.name, extra=err_lines)
[pipenv.exceptions.InstallError]: ['Collecting psycopg2-binary==2.7.7 (from -r /tmp/pipenv-xr5wpx_s-requirements/pipenv-449oezs3-requirement.txt (line 1))', '  Using cached https://files.pythonhosted.org/packages/dd/56/c22da10f5a725d61c58a185ec0f803aa2d384646ee8eb83d8ce88ed5edb1/psycopg2-binary-2.7.7.tar.gz', '    Complete output from command python setup.py egg_info:', '    running egg_info', '    creating pip-egg-info/psycopg2_binary.egg-info', '    writing pip-egg-info/psycopg2_binary.egg-info/PKG-INFO', '    writing dependency_links to pip-egg-info/psycopg2_binary.egg-info/dependency_links.txt', '    writing top-level names to pip-egg-info/psycopg2_binary.egg-info/top_level.txt', "    writing manifest file 'pip-egg-info/psycopg2_binary.egg-info/SOURCES.txt'", '    ', '    Error: pg_config executable not found.', '    ', '    pg_config is required to build psycopg2 from source.  Please add the directory', '    containing pg_config to the $PATH or specify the full executable path with the', '    option:', '    ', '        python setup.py build_ext --pg-config /path/to/pg_config build ...', '    ', "    or with the pg_config option in 'setup.cfg'.", '    ', '    If you prefer to avoid building psycopg2 from source, please install the PyPI', "    'psycopg2-binary' package instead.", '    ', "    For further information please check the 'doc/src/install.rst' file (also at", '    <http://initd.org/psycopg/docs/install.html>).', '    ', '    ', '    ----------------------------------------']
[pipenv.exceptions.InstallError]: ['Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-3f1iqb1e/psycopg2-binary/']
ERROR: ERROR: Package installation failed...
ERROR: Service 'drillsqueak' failed to build: The command '/bin/sh -c pipenv install' returned a non-zero code: 1
```

I believe this is the clue we need from that: `Error: pg_config executable not found.`

Furthermore, I question the usefulness of the extra build targets (`drillsqueak`, `mecha3-dev`, `mechasqueak`). None of the SPARK config files referenced in `docker-compose.yml` exist (nor do they have any example files, we only have the one "global" example). The extra build tasks are all the exact same container, only changing the config file parameter. Out of the box, every single one will fail to run. Would it not make more sense for the config file to be a parameter given to docker-compose and passed down?

**Comment by theunkn0wn1** [comment]
yeah... the actual mecha build targets have been broken for some time, and they don't even work for me. at this point i honestly question the usefulness of building mecha3 in a container at all.

File: docker-compose.yml [hunk 2]
```diff
@@ -54,7 +54,7 @@ services:
         target: /mechasqueak/certs/
     build: .
     image: mecha3
-    command: ["pipenv","run","python","main.py","--config", "develop.json"]
+    command: ["pipenv","run","python","main.py","--config", "develop.toml"]
```
Comment (Eearslya) [inline docker-compose.yml:57]: [Error] Oh, and we don't have a `main.py` anymore.

**Comment by theunkn0wn1** [comment]
Alright, at this point i seriously question why we should maintain this stuff itself.

Our production environment isn't going to use these files, our CI environment doesn't use these files, the only value they have is for Developers. Frankly, even in this roll its purely for convenience and as such its value is very limited.

Thus i propose this:
Why don't we just drop the docker stuff from the repository entirely?

cc @shatteredbeam @Eearslya

**Comment by Eearslya** [comment]
No objections.

**Pull request closed without merging**